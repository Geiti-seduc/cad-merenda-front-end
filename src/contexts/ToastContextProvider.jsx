// 1. Import necessary React modules
import React, { useRef, createContext } from 'react';
import PropTypes from 'prop-types';
import { Toast } from 'primereact/toast';

export const ToastContext = createContext();

function ToastContextProvider({ children }) {
  const toast = useRef(null);
  const showToast = (severity, summary, detail) => {
    toast.current.show({
      severity, summary, detail, life: 3000,
    });
  };

  const showSuccessToast = (message, title) => {
    toast.current.show({
      severity: 'success', summary: title, detail: message, life: 3000,
    });
  };

  const showErrorToast = (message, title) => {
    toast.current.show({
      severity: 'error', summary: title, detail: message, life: 3000,
    });
  };

  const showInfoToast = (message, title) => {
    toast.current.show({
      severity: 'info', summary: title, detail: message, life: 3000,
    });
  };

  const showContentToast = (severity, content) => {
    toast.current.show({
      severity, content, life: 3000,
    });
  };

  const contextValue = React.useMemo(() => ({
    showToast,
    showSuccessToast,
    showErrorToast,
    showInfoToast,
    showContentToast,
  }), [showToast]);

  return (
    <ToastContext.Provider value={contextValue}>
      <Toast ref={toast} />
      {children}
    </ToastContext.Provider>
  );
}

export const useToast = () => {
  const context = React.useContext(ToastContext);

  if (!context) {
    throw new Error('useToastContext must be used within a ToastContextProvider');
  }

  return context;
};

ToastContextProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default ToastContextProvider;
