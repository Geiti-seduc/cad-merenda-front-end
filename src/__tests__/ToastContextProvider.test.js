/** @jest-environment jsdom */
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import ToastContextProvider from '../contexts/ToastContextProvider';

describe('ToastContextProvider', () => {
  it('should render ToastContextProvider component with child component', () => {
    const ChildComponent = () => <div>Child Component</div>;
    const { asFragment } = render(
      <ToastContextProvider>
        <ChildComponent />
      </ToastContextProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
