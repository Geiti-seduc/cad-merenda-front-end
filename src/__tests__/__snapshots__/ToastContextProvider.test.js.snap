// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastContextProvider should render ToastContextProvider component 1`] = `
<DocumentFragment>
  <div
    class="p-toast p-component p-toast-top-right p-ripple-disabled"
    data-pc-name="toast"
    data-pc-section="root"
    style="position: fixed; top: 20px; right: 20px;"
  >
    <div />
  </div>
</DocumentFragment>
`;

exports[`ToastContextProvider should render ToastContextProvider component with child component 1`] = `
<DocumentFragment>
  <div
    class="p-toast p-component p-toast-top-right p-ripple-disabled"
    data-pc-name="toast"
    data-pc-section="root"
    style="position: fixed; top: 20px; right: 20px;"
  >
    <div />
  </div>
  <div>
    Child Component
  </div>
</DocumentFragment>
`;
